sklearn/cross_validation.py

```diff
@@ -412,6 +412,16 @@ def __init__(self, labels, n_folds=3, shuffle=False, random_state=None):
                      " than the number of labels: {1}.").format(n_folds,
                                                                 n_labels))
 
+        if shuffle:
+            # In case of ties in label weights, label names are indirectly
+            # used to assign samples to folds. When shuffle=True, label names
+            # are randomized to obtain random fold assigments.
+            rng = check_random_state(self.random_state)
+            unique_labels = np.arange(n_labels, dtype=np.int)
```

Comment:
It would be better never to use Python dtype (that have an implicitly defined size / precision level) but instead use numpy dtypes only, e.g. `np.intp` in this case: `np.intp` is the smallest integer dtype that is big enough to index any array.